File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
import sys

def solve():
    for line in sys.stdin:
        if not line: return
        n = int(line.rstrip())
        if n == -1: return
        
        nsum = 0
        nlist = []
        for i in range(1, n):
            if n % i == 0:
                nsum += i
                nlist.append(i)
        
        if n == nsum:
            print(n, "= ", end="")
            print(" + ".join(map(str, nlist)))
        else:
            print(n, "is NOT perfect.")
            
if __name__ == "__main__":
    solve()

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.cc
#include <iostream>
#include <vector>
using namespace std;

int main(void) {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);

	int n = 0;
	while (cin >> n && n != -1) {
		vector<int> nlist;
		int nsum = 0;
		for (int i = 1; i < n; i++) {
			if (n % i == 0) {
				nsum += i;
				nlist.push_back(i);
			}
		}

		if (n == nsum) {
			cout << n << " = ";
			for (int i = 0; i < nlist.size(); i++) {
				cout << nlist[i] << (i + 1 == nlist.size() ? "" : " + ");
			}
			cout << "\n";
		}
		else {
			cout << n << " is NOT perfect." << "\n";
		}
	}

	return 0;
}